src/lib.rs

```diff
@@ -42,100 +46,100 @@ pub use value::{BitSequence, Composite, Primitive, Value, ValueDef, Variant};
+    ///
+    /// let value2 = Value::unnamed_variant("B", [
+    ///     Value::u128(123),
+    ///     Value::bool(true),
+    /// ]);
+    /// let foo2: Foo = scale_value::serde::from_value(value2).unwrap();
+    /// assert_eq!(foo2, Foo::B(123, true));
+    /// ```
+    pub fn from_value<'de, Ctx, T: serde::Deserialize<'de>>(
+        value: crate::Value<Ctx>,
+    ) -> Result<T, DeserializerError> {
+        T::deserialize(value)
+    }
 
-	/// Attempt to convert some type into a [`crate::Value`] via serde.
-	///
-	/// # Examples
-	///
-	/// Convert a built-in array of values into a [`crate::Value`]:
-	///
-	/// ```rust
-	/// use scale_value::Value;
-	///
-	/// let arr = [1u8, 2u8, 3u8];
-	///
-	/// let val = scale_value::serde::to_value(arr).unwrap();
-	/// assert_eq!(val, Value::unnamed_composite([
-	///     Value::u128(1),
-	///     Value::u128(2),
-	///     Value::u128(3),
-	/// ]));
-	/// ```
-	///
-	/// Converting some custom type to a [`crate::Value`]:
-	///
-	/// ```rust
-	/// use scale_value::Value;
-	/// use serde::{ Serialize, Deserialize };
-	///
-	/// #[derive(Serialize, Deserialize, PartialEq, Debug)]
-	/// enum Foo {
-	///     A { is_valid: bool, name: String },
-	///     B(u8, bool)
-	/// }
-	///
-	/// let foo = Foo::A { is_valid: true, name: "James".into() };
-	///
-	/// let value = scale_value::serde::to_value(foo).unwrap();
-	/// assert_eq!(value, Value::named_variant("A", [
-	///     ("is_valid", Value::bool(true)),
-	///     ("name", Value::string("James")),
-	/// ]));
-	/// ```
-	pub fn to_value<T: serde::Serialize>(ty: T) -> Result<crate::Value<()>, SerializerError> {
-		ty.serialize(ValueSerializer)
-	}
+    /// Attempt to convert some type into a [`crate::Value`] via serde.
+    ///
+    /// # Examples
+    ///
+    /// Convert a built-in array of values into a [`crate::Value`]:
+    ///
+    /// ```rust
+    /// use scale_value::Value;
+    ///
+    /// let arr = [1u8, 2u8, 3u8];
+    ///
+    /// let val = scale_value::serde::to_value(arr).unwrap();
+    /// assert_eq!(val, Value::unnamed_composite([
+    ///     Value::u128(1),
+    ///     Value::u128(2),
+    ///     Value::u128(3),
+    /// ]));
+    /// ```
+    ///
+    /// Converting some custom type to a [`crate::Value`]:
+    ///
+    /// ```rust
+    /// use scale_value::Value;
+    /// use serde::{ Serialize, Deserialize };
+    ///
+    /// #[derive(Serialize, Deserialize, PartialEq, Debug)]
+    /// enum Foo {
+    ///     A { is_valid: bool, name: String },
+    ///     B(u8, bool)
+    /// }
+    ///
+    /// let foo = Foo::A { is_valid: true, name: "James".into() };
+    ///
+    /// let value = scale_value::serde::to_value(foo).unwrap();
+    /// assert_eq!(value, Value::named_variant("A", [
+    ///     ("is_valid", Value::bool(true)),
+    ///     ("name", Value::string("James")),
+    /// ]));
+    /// ```
+    pub fn to_value<T: serde::Serialize>(ty: T) -> Result<crate::Value<()>, SerializerError> {
+        ty.serialize(ValueSerializer)
+    }
```

Comment:
nit: there's an `Exmaple` -> `Example`; `encode and desome` -> encode and decode` below this

Reply:
Fixed!